src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerDifferenceService.kt

```diff
@@ -73,13 +74,23 @@ class PrisonerDifferenceService(
     }
   }
 
+  private fun prisonerHasChanged(nomsNumber: String, prisoner: Prisoner): Boolean =
+    with(UUID.randomUUID().toString()) {
+      updateHash(nomsNumber, prisoner, this).run { didUpdateHash(nomsNumber, this@with) }
+    }
+
```

Comment:
update the hash then check if the last update was done with the unique UUID we just generated, so even if JDBC reports a row count of 0 this should double check if we were indeed successful in updating the row.

Reply:
So if this is a bug around the number of returned rows it would be nice to prove that... Could we log the results of updateHash and didUpdateHash so we can see the bug in action?

Reply:
Yes, a good idea. Can we merge this to prove it works in T3 than I'll do another PR with the extra logging

Reply:
Interesting! Maybe log something in the event that the row was indeed updated and count was 0?

Reply:
yep